Core/Src/main.c

```diff
@@ -87,19 +95,27 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
     /* USER CODE END WHILE */
-
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+	  HAL_SPI_TransmitReceive_IT(&hspi1, tx_buf, rx_buf, sizeof(rx_buf));
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+	  uint16_t scaled_ADC_val = ((*rx_buf) & KEEP_10_LSB) * (MAX_TIMER_VALUE/MAX_ADC_VALUE); // drop 6 MSB and scales adc value to be out of 64000 instead of 1023
+	  uint16_t max_timer_val_for_high_output = (uint8_t)(0.05 * MAX_TIMER_VALUE) + 0.05 * scaled_ADC_val; //scales ADC digital value again to be between 5-10% of 65535
+	  __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, max_timer_val_for_high_output); //set compare register
     /* USER CODE BEGIN 3 */
   }
   /* USER CODE END 3 */
+  HAL_TIM_PWM_DeInit(&htim1);
```

Comment:
Can remove this. It has a good change of never running as this program never exits the while loop.